### PR TITLE
Refactor Message Templates (just Alert and Confirmation)

### PR DESCRIPTION
**Pull request by e7mac**
What              | Where/Who
------------------|----------------------------------------
JIRA Issue        | [JIRA_TICKET_ID](https://leanplum.atlassian.net/browse/JIRA_TICKET_ID)
People Involved   | @milos1290

[Notes on PR descriptions](https://leanplum.atlassian.net/wiki/spaces/PR/pages/288424408/Creating+a+GitHub+Pull+Request)

### CODE REVIEW

File: Leanplum-SDK/Classes/MessageTemplates/LPBaseMessageTemplate.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#import "LPMessageTemplateConstants.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface LPBaseMessageTemplate : NSObject
+
+-(void)defineActionWithContexts:(NSMutableArray *)contexts;
+- (UIViewController *)visibleViewController;
+
+@property  (nonatomic, strong) NSMutableArray *contexts;
```
Comment (milos1290) [inline Leanplum-SDK/Classes/MessageTemplates/LPBaseMessageTemplate.h]: declare properties first

File: Leanplum-SDK/Classes/MessageTemplates/LPBaseMessageTemplate.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#import <Foundation/Foundation.h>
+#import "Leanplum.h"
+#import "LPMessageTemplateConstants.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface LPBaseMessageTemplate : NSObject
+
+-(void)defineActionWithContexts:(NSMutableArray *)contexts;
+- (UIViewController *)visibleViewController;
```
Comment (milos1290) [inline Leanplum-SDK/Classes/MessageTemplates/LPBaseMessageTemplate.h]: formatting

File: Leanplum-SDK/Classes/MessageTemplates/LPBaseMessageTemplate.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+@property  (nonatomic, strong) NSMutableArray *contexts;
+
+
```
Comment (milos1290) [inline Leanplum-SDK/Classes/MessageTemplates/LPBaseMessageTemplate.h]: remove empty line

File: Leanplum-SDK/Classes/LPMessageTemplates.m [hunk 4]
```diff
@@ -230,89 +131,10 @@ + (UIViewController *)visibleViewController
...
-                     [self->_contexts addObject:context];
-                     return YES;
-                 }
-                 @catch (NSException *exception) {
-                     LOG_LP_MESSAGE_EXCEPTION;
-                     return NO;
-                 }
-             }];
-    
+    [[[LPAlertMessageTemplate alloc] init] defineActionWithContexts:_contexts];
```
Comment (milos1290) [inline Leanplum-SDK/Classes/LPMessageTemplates.m]: maybe `[LPAlertMessageTemplate new] defineActionWithContexts:_contexts];` ?

Comment (e7mac) [reply Leanplum-SDK/Classes/LPMessageTemplates.m]: `[Class new]` is conventionally old Obj-C so I don't use that

**Pull request merged**